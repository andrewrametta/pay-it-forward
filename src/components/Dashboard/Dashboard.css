.dashboard-container {
  margin-bottom: 100px;
}

p .p-citystate {
  color: lightgrey;
  font-size: 12px;
}

div.donation-dashboard-img {
  text-align: center;
}

li.items-div-container {
  margin: 10px;
  background: white;
  padding: 10px;
  box-shadow: 2px 2px 6px 0 rgba(41, 2, 80, 0.1);
  border-radius: 10px;
}

.ul-items {
  text-align: center;
}

img.img-donation {
  width: 100%;
  height: 100%;
  padding: 5px;
}

@media all and (min-width: 400px) {
  .ul-items {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
  }

  img.img-donation {
    width: 150px;
    height: 150px;
    padding: 5px;
  }
}
