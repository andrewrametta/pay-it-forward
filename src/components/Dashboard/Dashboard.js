import React, { useContext, useEffect, useState } from "react";
import { Image } from "cloudinary-react";
import AppContext from "../../AppContext";
import { Link } from "react-router-dom";
import AuthAPIService from "../../services/auth-api-service";
import "./Dashboard.css";

export default function Dashboard(props) {
  const [error, setError] = useState("");
  const { items, setItems } = useContext(AppContext);

  useEffect(() => {
    AuthAPIService.getItems()
      .then((donations) => {
        console.log(donations);
        setItems(donations);
      })
      .catch((error) => {
        setError(error);
      });
  }, [setItems, setError]);

  return (
    <div className="dashboard-container">
      <h1>Donations Available</h1>
      <div className="dashboard-item-container">
        {error && <h2>error</h2>}
        <ul className="ul-items">
          {items
            .sort((a, b) => a.id - b.id)
            .map((item, indx) => (
              <Link key={indx} to={`donation/${item.id}`}>
                <li key={indx} className="items-div-container">
                  <div className="donation-dashboard-img">
                    {item.item_url ? (
                      <Image
                        className="img-donation"
                        cloudName="hq1rpt94r"
                        publicId={`${item.item_url}`}
                        width="275"
                        height="275"
                        crop="fill"
                      />
                    ) : null}
                  </div>

                  <h3>{item.title}</h3>
                  <p>{item.status}</p>
                  <p className="p-citystate">
                    {item.city}, {item.state}
                  </p>
                </li>
              </Link>
            ))}
        </ul>
      </div>
    </div>
  );
}
