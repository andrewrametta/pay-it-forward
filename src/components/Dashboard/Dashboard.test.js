import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Dashboard from "./Dashboard";
// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AppContext from "../../AppContext";

configure({ adapter: new Adapter() });

describe(`Dashboard component`, () => {
  it.only("renders the complete page", () => {
    const wrapper = shallow(
      <AppContext.Provider>
        <Dashboard />
      </AppContext.Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
