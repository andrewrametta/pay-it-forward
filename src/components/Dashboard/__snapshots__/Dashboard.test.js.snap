// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component renders the complete page 1`] = `<Dashboard />`;
